src/folderlistmodel.cpp

```diff
@@ -364,6 +432,23 @@ FolderListModel::FolderStandardType FolderListModel::folderTypeFromId(const QMai
     return NormalFolder;
 }
 
+bool FolderListModel::isStandardFolder(const QMailFolderId &id) const
+{
+    FolderStandardType folderType = folderTypeFromId(id);
+    return folderType == InboxFolder || folderType == DraftsFolder
+            || folderType == SentFolder || folderType == TrashFolder || folderType == OutboxFolder;
+}
+
+FolderListModel::FolderItem* FolderListModel::createFolderItem(QModelIndex idx, QMailFolderId mailFolderId,
```

Comment:
QModelIndex, QMailFolderId and QMailMessageKey would all be better passed by const &.